File: Solutions/Java/1 - TwoSum.java
import java.util.HashMap;

public class Main {

    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        Integer target = 9;

        Solution solution = new Solution();
        int[] sol = solution.twoSum(nums, target);
        System.out.println("first : " + sol[0] + " second : " + sol[1]);
    }
}

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> valueByPosition = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (null != valueByPosition.get(nums[i])) {
                return new int[]{valueByPosition.get(nums[i]), i};
            } else {
                valueByPosition.put(target - nums[i], i);
            }
        }
        return new int[2];
    }
}

File: Solutions/Python/1 - TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashmap = {}
        for i in range(len(nums)):
            if hashmap.get(nums[i]) is not None:
                return [hashmap.get(nums[i]), i]
            else:
                hashmap[target - nums[i]] = i
        return None

nums = [2, 7, 11, 15]
target = 9
print(Solution().twoSum(nums, target))